File: sdkwork-backend-sdk-go/api/vip_level_benefit.go
package api

import (
    "fmt"
    sdktypes "github.com/sdkwork/backend-sdk/types"
    sdkhttp "github.com/sdkwork/backend-sdk/http"
)

type VipLevelBenefitApi struct {
    client *sdkhttp.Client
}

func NewVipLevelBenefitApi(client *sdkhttp.Client) *VipLevelBenefitApi {
    return &VipLevelBenefitApi{client: client}
}

// Update an existing VIP level benefit
func (a *VipLevelBenefitApi) Update(body sdktypes.PlusVipLevelBenefitForm) (sdktypes.PlusApiResultPlusVipLevelBenefitVO, error) {
    raw, err := a.client.Put(BackendApiPath("/vip/level/benefit"), body, nil, nil, "")
    if err != nil {
        var zero sdktypes.PlusApiResultPlusVipLevelBenefitVO
        return zero, err
    }
    return decodeResult[sdktypes.PlusApiResultPlusVipLevelBenefitVO](raw)
}

// Create a new VIP level benefit
func (a *VipLevelBenefitApi) Create(body sdktypes.PlusVipLevelBenefitForm) (sdktypes.PlusApiResultPlusVipLevelBenefitVO, error) {
    raw, err := a.client.Post(BackendApiPath("/vip/level/benefit"), body, nil, nil, "")
    if err != nil {
        var zero sdktypes.PlusApiResultPlusVipLevelBenefitVO
        return zero, err
    }
    return decodeResult[sdktypes.PlusApiResultPlusVipLevelBenefitVO](raw)
}

// Get VIP level benefits by page
func (a *VipLevelBenefitApi) ListByPage(body *sdktypes.QueryListForm, query map[string]interface{}) (sdktypes.PlusApiResultPagePlusVipLevelBenefitVO, error) {
    raw, err := a.client.Post(BackendApiPath("/vip/level/benefit/list"), body, query, nil, "")
    if err != nil {
        var zero sdktypes.PlusApiResultPagePlusVipLevelBenefitVO
        return zero, err
    }
    return decodeResult[sdktypes.PlusApiResultPagePlusVipLevelBenefitVO](raw)
}

// Get all VIP level benefits
func (a *VipLevelBenefitApi) ListAllEntities(body *sdktypes.QueryListForm) (sdktypes.PlusApiResultListPlusVipLevelBenefitVO, error) {
    raw, err := a.client.Post(BackendApiPath("/vip/level/benefit/list/all"), body, nil, nil, "")
    if err != nil {
        var zero sdktypes.PlusApiResultListPlusVipLevelBenefitVO
        return zero, err
    }
    return decodeResult[sdktypes.PlusApiResultListPlusVipLevelBenefitVO](raw)
}

// Get a VIP level benefit by ID
func (a *VipLevelBenefitApi) GetById(id string) (sdktypes.PlusApiResultPlusVipLevelBenefitVO, error) {
    raw, err := a.client.Get(BackendApiPath(fmt.Sprintf("/vip/level/benefit/%s", id)), nil, nil)
    if err != nil {
        var zero sdktypes.PlusApiResultPlusVipLevelBenefitVO
        return zero, err
    }
    return decodeResult[sdktypes.PlusApiResultPlusVipLevelBenefitVO](raw)
}

// Delete a VIP level benefit
func (a *VipLevelBenefitApi) Delete(id string) (sdktypes.PlusApiResultBoolean, error) {
    raw, err := a.client.Delete(BackendApiPath(fmt.Sprintf("/vip/level/benefit/%s", id)), nil, nil)
    if err != nil {
        var zero sdktypes.PlusApiResultBoolean
        return zero, err
    }
    return decodeResult[sdktypes.PlusApiResultBoolean](raw)
}

File: sdkwork-backend-sdk-typescript/src/api/vip-level-benefit.ts
import { backendApiPath } from './paths';
import type { HttpClient } from '../http/client';
import type { QueryParams } from '../types/common';
import type { PlusApiResultBoolean, PlusApiResultListPlusVipLevelBenefitVO, PlusApiResultPagePlusVipLevelBenefitVO, PlusApiResultPlusVipLevelBenefitVO, PlusVipLevelBenefitForm, QueryListForm } from '../types';


export class VipLevelBenefitApi {
  private client: HttpClient;
  
  constructor(client: HttpClient) { 
    this.client = client; 
  }

/** Update an existing VIP level benefit */
  async update(body: PlusVipLevelBenefitForm): Promise<PlusApiResultPlusVipLevelBenefitVO> {
    return this.client.put<PlusApiResultPlusVipLevelBenefitVO>(backendApiPath(`/vip/level/benefit`), body);
  }

/** Create a new VIP level benefit */
  async create(body: PlusVipLevelBenefitForm): Promise<PlusApiResultPlusVipLevelBenefitVO> {
    return this.client.post<PlusApiResultPlusVipLevelBenefitVO>(backendApiPath(`/vip/level/benefit`), body);
  }

/** Get VIP level benefits by page */
  async listByPage(body?: QueryListForm, params?: QueryParams): Promise<PlusApiResultPagePlusVipLevelBenefitVO> {
    return this.client.post<PlusApiResultPagePlusVipLevelBenefitVO>(backendApiPath(`/vip/level/benefit/list`), body, params);
  }

/** Get all VIP level benefits */
  async listAllEntities(body?: QueryListForm): Promise<PlusApiResultListPlusVipLevelBenefitVO> {
    return this.client.post<PlusApiResultListPlusVipLevelBenefitVO>(backendApiPath(`/vip/level/benefit/list/all`), body);
  }

/** Get a VIP level benefit by ID */
  async getById(id: string | number): Promise<PlusApiResultPlusVipLevelBenefitVO> {
    return this.client.get<PlusApiResultPlusVipLevelBenefitVO>(backendApiPath(`/vip/level/benefit/${id}`));
  }

/** Delete a VIP level benefit */
  async delete(id: string | number): Promise<PlusApiResultBoolean> {
    return this.client.delete<PlusApiResultBoolean>(backendApiPath(`/vip/level/benefit/${id}`));
  }
}

export function createVipLevelBenefitApi(client: HttpClient): VipLevelBenefitApi {
  return new VipLevelBenefitApi(client);
}
